src/components/Index.vue

```diff
@@ -1,147 +1,56 @@
 <template>
-  <div :class="suit()">
+  <!-- Index is an empty component that will hold other widgets -->
+  <div>
     <slot></slot>
   </div>
 </template>
 
 <script>
-import { createFromAlgoliaCredentials } from '../store';
-import algoliaComponent from '../component';
 import instantsearch from 'instantsearch.js/es/';
 
 export default {
-  mixins: [algoliaComponent],
+  provide() {
+    this.instantSearchInstance = instantsearch({
+      appId: this.appId,
+      apiKey: this.apiKey,
+      indexName: this.indexName,
+    });
```

Comment:
the other props also need to be given here, and it needs to "restart" when they change, like in `algoliaComponent` I think